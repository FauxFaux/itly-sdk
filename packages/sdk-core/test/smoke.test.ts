/* eslint-disable import/no-unresolved, global-require */
/* eslint-disable import/extensions */
import itly from '../dist/index';
import CustomDestination from './src/CustomDestination';
import CustomLogger from './src/CustomLogger';

const testSchemas = require('./schemas/basic-schema.json');

test('should load, track events, and validate schema', () => {
  const spy = jest.spyOn(console, 'log');

  const userId = 'test-user-id';

  itly.load({
    environment: 'production',
    schemas: testSchemas,
    destinations: [
      new CustomDestination(),
    ],
    context: {
      requiredString: 'A required string',
      optionalEnum: 'Value 1',
    },
    logger: new CustomLogger(),
  });

  itly.track(userId, 'Event No Properties', {}, '26af925a-be3a-40e5-947d-33da66a5352f', '1-0-0');

  itly.track(
    userId,
    'Event With All Properties',
    {
      requiredString: 'A required string',
      requiredNumber: 2.0,
      requiredInteger: 42,
      requiredEnum: 'Enum1',
      requiredBoolean: false,
      requiredConst: 'some-const-value',
      requiredArray: ['required', 'array'],
      optionalString: 'I\'m optional!',
    },
    '311ba144-8532-4474-a9bd-8b430625e29a',
    '1-0-0',
  );

  try {
    itly.track(
      userId,
      'EventMaxIntForTest',
      {
        intMax10: 20,
      },
      'aa0f08ac-8928-4569-a524-c1699e7da6f4',
      '1-0-0',
    );
  } catch (e) {
    // do nothing
  }

  expect(spy.mock.calls).toMatchSnapshot();

  spy.mockRestore();
});
