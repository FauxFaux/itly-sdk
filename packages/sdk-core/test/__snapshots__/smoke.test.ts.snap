// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load, track events, and validate schema 1`] = `
Array [
  Array [
    "CustomDestination: ",
    "init()",
  ],
  Array [
    "CustomLogger.debug:",
    "[Itly] Tracking event \\"Event No Properties\\"",
  ],
  Array [
    "CustomDestination: ",
    "track() userId='test-user-id' event='Event No Properties' properties={}",
  ],
  Array [
    "CustomLogger.debug:",
    "[Itly] Tracking event \\"Event With All Properties\\"",
  ],
  Array [
    "CustomDestination: ",
    "track() userId='test-user-id' event='Event With All Properties' properties={\\"requiredString\\":\\"A required string\\",\\"requiredNumber\\":2,\\"requiredInteger\\":42,\\"requiredEnum\\":\\"Enum1\\",\\"requiredBoolean\\":false,\\"requiredConst\\":\\"some-const-value\\",\\"requiredArray\\":[\\"required\\",\\"array\\"],\\"optionalString\\":\\"I'm optional!\\"}",
  ],
  Array [
    "CustomLogger.debug:",
    "[Itly] Tracking event \\"EventMaxIntForTest\\"",
  ],
  Array [
    "CustomLogger.error::",
    "Passed in EventMaxIntForTest properties did not validate against your tracking plan. \`properties.intMax10\` should be <= 10.",
  ],
  Array [
    "CustomDestination: ",
    "track() userId='test-user-id' event='EventMaxIntForTest' properties={\\"intMax10\\":20}",
  ],
]
`;
